sentry-core/src/test/java/io/sentry/core/transport/HttpTransportTest.kt

```diff
@@ -32,6 +33,7 @@ class HttpTransportTest {
         var readTimeout = 500
         var bypassSecurity = false
         val connection = mock<HttpURLConnection>()
+        val currentDateProvider = mock<ICurrentDateProvider>()
```

Comment:
Instead of mocking this all the time, we could implement a `TestCurrentDateProvider` that always returns the same time like `new Date(0)` and use it here. Makes the code cleaner IMO.

Reply:
we'd need to have more than 1 fake though, as tests would need to return also values different than `Date(0)` eg if `isRetryAfter` returns `false` when the time has passed.
I think it's not a good idea and also not sure how to test this use case if the return is always the same?

what's your LOGAF here?